.github/workflows/scripts/build_linux_artifacts.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+set -e
+
+REPODIR=$(git rev-parse --show-toplevel)
+
+TAG=${TAG:-$(git log -n1 --pretty=%h)}
+OUTDIR=${OUTDIR:-"${REPODIR}/artifacts"}
+
+TARCMD=tar
+if [[ $(uname -s) == "Darwin" ]]; then
+    # When building linux artifacts from darwin, gtar is required.
+    TARCMD="gtar"
+fi
+
+TAROPTS=("--owner=root" "--group=root")
+
+TMPDIR=$(mktemp -d)
+cleanup() {
+    rm -rf "${TMPDIR}"
+}
+trap cleanup EXIT
+
+
+copy_binary_from_multiarch_tar() {
+    local arch=$1
+    local binary=$2
+    local destdir=$3
+
+    local srcpath="/opt/spire/bin/${binary}"
+    local destpath="${destdir}/${binary}"
+    local ocidir="ocidir://${TMPDIR}/${arch}/oci/${binary}"
+    local imagetar="${REPODIR}/${binary}-image.tar"
+    local platform="linux/${arch}"
+
+    echo "Importing multiarch image ${imagetar}..."
+    regctl image import "${ocidir}" "${imagetar}"
+
+    echo "Copying ${srcpath} for platform ${platform}..."
+    regctl image get-file "${ocidir}" "${srcpath}" "${destpath}" -p "${platform}"
+
+    # file does not retain permission bits, so fix up the executable bit.
+    chmod +x "${destpath}"
+}
+
+build_artifact() {
+    ARCH="$1"
```

Comment:
nit: for readability, many variable names in this function can be lowercased to signify they are local to this function and not exported into the environment

Reply:
i tried to minimize the changes to this block of code for the sake of review ease (since it closely matches the existing scripts). Happy to change this though.

Reply:
Let's leave it for now, can always clean up later.